store/loader.go

```diff
@@ -0,0 +1,224 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/metrics"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/eigenda"
+	eigendav2 "github.com/Layr-Labs/eigenda-proxy/store/generated_key/eigenda_v2"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore"
+	"github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/redis"
+	"github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/s3"
+	"github.com/Layr-Labs/eigenda-proxy/verify/v1"
+	"github.com/Layr-Labs/eigenda/api/clients"
+	clients_v2 "github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	"github.com/rs/zerolog/log"
+
+	"github.com/Layr-Labs/eigenda/common/geth"
+	eigenda_eth "github.com/Layr-Labs/eigenda/core/eth"
+	"github.com/Layr-Labs/eigenda/encoding/kzg"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	geth_common "github.com/ethereum/go-ethereum/common"
+)
+
+// StoreLoader centralizes dependency initialization
+// It ensures proper typing and avoids redundant logic scattered across functions.
+type StoreLoader struct {
```

Comment:
does this actually "load" anything? it more so feels like a builder to me, but maybe I'm misunderstanding how its used.

Reply:
[df44d5f](https://github.com/Layr-Labs/eigenda-proxy/pull/296/commits/df44d5f3bdcbdd76a5eaf2d5014cd4134bf88573)

Reply:
agree